subsys/net/ip/tcp.c

```diff
@@ -2239,8 +2252,11 @@ NET_CONN_CB(tcp_syn_rcvd)
 	struct sockaddr_ptr pkt_src_addr;
 	struct sockaddr local_addr;
 	struct sockaddr remote_addr;
+	enum net_verdict ret = NET_DROP;
```

Comment:
Please rename this to `verdict` as there is also `ret` variable in `if (NET_TCP_FLAGS(tcp_hdr) & NET_TCP_ACK) {` branch. The `ret` in that func will not return proper return value to net_core.c which will then cause the "already freed" error that I saw earlier.